armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -0,0 +1,492 @@
+# Copyright 2019 TerraPower, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""enable component-wise axial expansion for assemblies and/or a reactor"""
+
+from statistics import mean
+from numpy import array
+from armi import runLog
+from armi.reactor.flags import Flags
+
+
+class AxialExpansionChanger:
+    """
+    Axially expand or contract assemblies or an entire core.
+
+    Useful for fuel performance, thermal expansion, reactivity coefficients, etc.
+    """
+
+    def __init__(self, converterSettings: dict):
+        """
+        Build an axial expansion converter.
+
+        Parameters
+        ----------
+        converterSettings : dict
+            A set of str, value settings used in mesh conversion. Required
+            settings are implementation specific.
+        """
+        self._converterSettings = converterSettings
+        self.linked = None
+        self.expansionData = None
+
+    def setAssembly(self, a):
+        """set the armi assembly to be changed and init expansion data class for assembly
+
+        Parameters
+        ----------
+        a
```

Comment:
I highly recommend that we specify the types of the parameters being requested in the docstrings and as a type hint for all of the public methods that are implemented. We want to go this direction of having better quality doc strings and it's clearly lacking in most of the code base.

For the type hint, you can have `from armi.reactor.assrmblies import Assembly` at the top and then change this to: `setAssembly(self, a: Assembly):`

Additionally for the docstrings you can change this to

```
 a : :py:class:`Assembly <armi.reactor.assemblies.Assembly>` object. 

```

Reply:
Addressed.

https://github.com/terrapower/armi/pull/606/commits/b35802979a71dbb8f69c0b66baee1ae8bad9fa35